data/transactions/logic/eval.go

```diff
@@ -144,6 +155,22 @@ type LedgerForLogic interface {
 	GetDelta(txn *transactions.Transaction) (evalDelta basics.EvalDelta, err error)
 }
 
+// EvalSideEffects contains data returned from evaluation
+type EvalSideEffects struct {
+	scratchSpace scratchSpace
+}
+
+// GetScratchValue loads and clones a stackValue
+// The value is cloned so the original bytes are protected from changes
+func (se *EvalSideEffects) GetScratchValue(stackPos uint8) stackValue {
+	return se.scratchSpace[stackPos].clone()
```

Comment:
```suggestion
	return se.scratchSpace[scratchPos].clone()
```

nit: scratch is not stack